### PR TITLE
rootfs image installation/download progress display

### PR DESCRIPTION
**Pull request by bboozzoo**
Add support for installation progress display when meder is used in `-rootfs` mode. Demo:
https://asciinema.org/a/4cp2bwmdg5whlqywk7wyst3bi

@kacf @pasinskim @GregorioDiStefano

### CODE REVIEW

File: utils/progress.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+	for ; nowDots < thenDots; nowDots++ {
+		p.Out.Write([]byte("."))
+		if nowDots != 0 && (nowDots+1)%dotsPerLine == 0 {
+			// print percentage and size after each complete line
+			var s string
+			nowSize := (nowDots + 1) * perDot
+			nowSizekB := nowSize / 1024
+			if p.N == 0 {
+				s = fmt.Sprintf(" %v kB\n", nowSizekB)
```
Comment (kacf) [inline utils/progress.go]: Minor: Perhaps use `kiB`? This would make it completely unambiguous.

File: utils/progress.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	thenDots := then / perDot
+
+	for ; nowDots < thenDots; nowDots++ {
+		p.Out.Write([]byte("."))
+		if nowDots != 0 && (nowDots+1)%dotsPerLine == 0 {
+			// print percentage and size after each complete line
+			var s string
+			nowSize := (nowDots + 1) * perDot
+			nowSizekB := nowSize / 1024
+			if p.N == 0 {
```
Comment (kacf) [inline utils/progress.go]: I would perhaps do:

```
if p.N == 0 || then > p.N
```

to catch errors where the expected size is miscalculated. Unlikely, but possible. We might consider a warning as well.

File: utils/progress.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		if thenDots == 0 {
+			p.Out.Write([]byte("."))
+			left -= 1
+		}
+		// fill remaining space with spaces
+		for i := int64(0); i < left; i++ {
+			p.Out.Write([]byte(" "))
+		}
+		// print 100%, make sure we proper size suffix just in case
+		// we arre below 1kB with the whole size
```
Comment (kacf) [inline utils/progress.go]: `s/arre/are/`

File: utils/progress.go (hunk 1 above)
Comment (kacf) [inline utils/progress.go]: There are multiple places with `kB`, btw.

**Comment by kacf** [comment]
Nice demo, btw, didn't know about that site.

File: utils/progress.go (hunk 1 above)
Comment (bboozzoo) [inline utils/progress.go]: fixed

File: utils/progress.go (hunk 2 above)
Comment (bboozzoo) [inline utils/progress.go]: Fixed. Warning will be printed when `Write()` goes over declared size. The warning will appear only once.

**Comment by kacf** [comment]
This conflicts right now, but this is just because we had to temporarily revert #138 (see #140). Just leave it and it will un-conflict itself once we have fixed the tests.

**Comment by bboozzoo** [comment]
Rebased on top of master.

**Comment by kacf** [comment]
Running in Jenkins.

**Pull request merged**